From 5440bbc6b29f9fcabb689998633cc140ecd5110c Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:00:38 -0400
Subject: [fs] btrfs: commit transaction after setting label and features

Message-id: <1407179109-13714-152-git-send-email-sandeen@redhat.com>
Patchwork-id: 87088
O-Subject: [RHEL7.1 PATCH 151/422] btrfs: commit transaction after setting label and features
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Jeff Mahoney <jeffm@suse.com>

commit d0270aca88966641eb15306e9bd0c7ad15321440
Author: Jeff Mahoney <jeffm@suse.com>
Date:   Fri Feb 7 14:33:57 2014 +0100

    btrfs: commit transaction after setting label and features

    The set_fslabel ioctl uses btrfs_end_transaction, which means it's
    possible that the change will be lost if the system crashes, same for
    the newly set features. Let's use btrfs_commit_transaction instead.

    Signed-off-by: Jeff Mahoney <jeffm@suse.com>
    Signed-off-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/ioctl.c b/fs/btrfs/ioctl.c
index b013489..8e48b81 100644
--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -4525,7 +4525,7 @@ static int btrfs_ioctl_set_fslabel(struct file *file, void __user *arg)
  spin_lock(&root->fs_info->super_lock);
  strcpy(super_block->label, label);
  spin_unlock(&root->fs_info->super_lock);
- ret = btrfs_end_transaction(trans, root);
+ ret = btrfs_commit_transaction(trans, root);
 
 out_unlock:
  mnt_drop_write_file(file);
@@ -4689,7 +4689,7 @@ static int btrfs_ioctl_set_features(struct file *file, void __user *arg)
  btrfs_set_super_incompat_flags(super_block, newflags);
  spin_unlock(&root->fs_info->super_lock);
 
- return btrfs_end_transaction(trans, root);
+ return btrfs_commit_transaction(trans, root);
 }
 
 long btrfs_ioctl(struct file *file, unsigned int
-- 
1.7.1